File: 66-plus-one/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if(digits):
            n = len(digits)
            carry = 1
            for i in reversed(range(0,n)):
                val = digits[i] + carry
                if(val < 10):
                    digits[i] = val
                    return digits
                else:
                    carry = val // 10
                    digit = val % 10
                    digits[i] = digit
            if(carry):
                temp = [1]
                return temp + digits 
            return digits


        

File: 66-plus-one/plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let carry = 1, n = digits.length-1

    while(n >= 0 && carry){
        let val = digits[n] + carry
        let dig = val % 10

        digits[n] = dig

        carry = Math.floor(val/10)

        if(carry <= 0) break
        n--
    }

    if(carry == 1){
        digits = [1, ...digits]
    }

    return digits
        
    
};